File: backend/src/main/java/org/example/DTO/StatusEnum.java
package org.example.DTO;

import java.util.Arrays;

public enum StatusEnum {
    NEW("Новый"),
    PAID("Оплачен"),
    AWAITING_PAYMENT ("Ожидает оплаты"),
    COMPLETED("Завершен"),
    AWAITING_CONFIRMATION("Ожидает подтверждения"),
    ISSUED("Выдано");
    private final String name;

    StatusEnum(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public static StatusEnum fromName(String name) {
        return Arrays.stream(values())
                .filter(status -> status.name.equalsIgnoreCase(name))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("Неизвестный статус: " + name));
    }
}

File: frontend/app/src/main/java/com/savelyev/MobileApp/Api/DTO/Enum/StatusEnum.kt
package com.savelyev.MobileApp.Api.DTO.Enum

import com.fasterxml.jackson.annotation.JsonValue

enum class StatusEnum(private val status: String) {
    NEW("Новый"),
    ISSUED ("Выдано"),
    AWAITING_PAYMENT ("Ожидает оплаты"),
    PAID("Оплачен"),
    COMPLETED("Завершен"),
    AWAITING_CONFIRMATION("Ожидает подтверждения"),
    CANCELLED("Отменен");

    @JsonValue
    fun toValue(): String = status

    companion object {
        fun fromStatus(status: String): String {
            return try {
                valueOf(status).toValue()
            } catch (e: IllegalArgumentException) {
                "Неизвестный статус"
            }
        }
    }
}